docs: PEM-916 Documentation of Monitoring Packs (#1157)
spectrocloud/librarium: content/docs/06-integrations/00-prometheus-operator.md

<br />

#### Email Alerts

You can configure the Prometheus server to send email alerts to a set of contacts. Toggle the **Email Alerts** button to enable email alerting. Update the `alertmanager.config.receivers` settings with all the required email setting values.

<br />

```yaml
charts:
  kube-prometheus-stack:
    alertmanager:
      config:
        receivers:
          - name: email-alert
            email_configs:
              - to: <reciever>@<domainname>.com
                send_resolved: true
                from: <sender>@<domainname>.com
                smarthost: smtp.<domainname>.com:587
                auth_username: <sender>@<domainname>.com
                auth_identity: <sender>@<domainname>.com
                auth_password: <sender_passwd>
```

Refer to the [Prometheus Alertmanager Configuration](https://prometheus.io/docs/alerting/latest/configuration/) documentation to learn more about Alertmanager.

<br />

#### Grafana Ingress

You can enable an ingress endpoint for Grafana that will deploy an NGINX ingress controller. This feature can be used to enable HTTPS and require authentication for all Prometheus API requests. 

> By default, a service with a load balancer, exposing port 80 is created unless the ingress option is enabled. 

Reviewer: ```suggestion
If you do not enable the ingress option, then by default a service with a load balancer will be created that exposes port 80. 

```

Toggle the **Enable** button to enable the use of Ingress.

<br />

#### Thanos SideCar

> [Thanos](https://prometheus-operator.dev/docs/operator/thanos/) is an open-source system for running large-scale, distributed, and highly available Prometheus setups. Thanos allows Prometheus to store data for extended periods by enabling data to be stored in object storage, such as Amazon S3 or Google Cloud Storage, instead of a local disk. This enables Prometheus to scale horizontally without needing to worry about running out of local storage space.

Reviewer: ```suggestion
[Thanos](https://prometheus-operator.dev/docs/operator/thanos/) is an open-source system for running large-scale, distributed, and highly available Prometheus setups. Thanos allows Prometheus to store data for extended periods in object storage, such as Amazon S3 or Google Cloud Storage, instead of a local disk. This enables Prometheus to scale horizontally without the risk of using up local storage space.

```

Toggle the **Enable** button to enable the use of Thanos.

<br />

#### Object Store

Select the Thanos object storage type you will use. Review the `thanos.objstoreConfig` parameters to configure the use of object storage with Thanos. Refer to the [Thanos Object Storage](https://github.com/thanos-io/thanos/blob/main/docs/storage.md) documentation to learn more about how to configure each object storage.

<br />

```yaml
charts:
  kube-prometheus-stack:
    prometheus:
      prometheusSpec:
        thanos:
          objstoreConfig:
```

#### Thanos Ruler Object Store

> You can specify a different object storage to store the Thanos Ruler event data.  Defaults to the object storage specified for Thanos. Refer to the [Thanos Ruler](https://prometheus-operator.dev/docs/operator/thanos/?#thanos-ruler) resource to learn more.

Reviewer: ```suggestion
By default, Thanos Ruler event data is saved in object storage specified for Thanos, but you can specify a different object storage for event data.  Refer to the [Thanos Ruler](https://prometheus-operator.dev/docs/operator/thanos/?#thanos-ruler) resource to learn more.

```

<br />

#### Remote Monitoring

> You can configure the Prometheus server to accept metrics from Prometheus agents and become a centralized aggregation point for all Kubernetes metrics. Enabling this feature will expose the *prometheus-operator-prometheus* service's port 9090. Use the generated service URL to provide other Kubernetes clusters with the [Prometheus Agent](/integrations/prometheus-agent) installed so that cluster metrics can be forwarded to the Prometheus server. 

Reviewer: ```suggestion
You can configure the Prometheus server to accept metrics from Prometheus agents and become a centralized aggregation point for all Kubernetes metrics. Enabling this feature will expose port 9090 of the *prometheus-operator-prometheus* service. Use the generated service URL to provide other Kubernetes clusters with the [Prometheus Agent](/integrations/prometheus-agent) installed so that cluster metrics can be forwarded to the Prometheus server. 

```

> The remote monitoring feature is configured with defaults to help you consume this feature out-of-the-box. You can change any configuration related to remote monitoring to fine-tune settings for your environment.

Reviewer: ```suggestion
The remote monitoring feature is configured with defaults to help you consume this feature out-of-the-box. You can change any configuration related to remote monitoring to fine tune settings for your environment.

```

* [Prometheus](https://prometheus.io/)
* [Prometheus Alertmanager](https://www.prometheus.io/docs/alerting/latest/alertmanager/).
* [node-exporter](https://github.com/helm/charts/tree/master/stable/prometheus-node-exporter)
* [kube-state-metrics](https://github.com/helm/charts/tree/master/stable/kube-state-metrics)
* [Grafana](https://github.com/helm/charts/tree/master/stable/grafana)
* and the service monitors to scrape internal Kubernetes components.

</Tabs.TabPane>

<Tabs.TabPane tab="40.0.x" key="40.0.x">

## Prerequisites

* Kubernetes v1.16 or greater.

## Parameters

The Prometheus operator supports all the parameters exposed by the kube-prometheus-stack Helm Chart. Refer to the [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stackn) documentation for details.

The Prometheus Operator pack has one parameter you must initialize `grafana.adminPassword`:

<br />

```yaml
charts:
  kube-prometheus-stack:
    grafana:
      adminPassword: ""
```

Use the `grafana.adminPassword` parameter to assign a password to the Grafana admin user `admin`. 

> Additional parameters you should be aware can be found by expanding the **Presets** view of the pack. You can modify the preset settings during the profile creation process or the cluster deployment process when reviewing the cluster profile.

Reviewer: ```suggestion
Additional parameters you should be aware of can be found by expanding the **Presets** options. You can modify the preset settings when you create the profile or when you deploy the cluster and review the cluster profile.

```

<br />

#### Email Alerts

You can configure the Prometheus server to send email alerts to a set of contacts. Toggle the **Email Alerts** button to enable email alerting. Update the `alertmanager.config.receivers` settings with all the required email setting values.

<br />

```yaml
charts:
  kube-prometheus-stack:
    alertmanager:
      config:
        receivers:
          - name: email-alert
            email_configs:
              - to: <reciever>@<domainname>.com
                send_resolved: true
                from: <sender>@<domainname>.com
                smarthost: smtp.<domainname>.com:587
                auth_username: <sender>@<domainname>.com
                auth_identity: <sender>@<domainname>.com
                auth_password: <sender_passwd>
```

Refer to the [Prometheus Alertmanager Configuration](https://prometheus.io/docs/alerting/latest/configuration/) documentation to learn more about Alertmanager.

<br />

#### Grafana Ingress

You can enable an ingress endpoint for Grafana that will deploy an NGINX ingress controller. This feature can be used to enable HTTPS and require authentication for all Prometheus API requests. 

> By default, a service with a load balancer, exposing port 80 is created unless the ingress option is enabled. 

Reviewer: ```suggestion
If you do not enable the ingress option, by default a service with a load balancer will be created that exposes port 80.

```

Toggle the **Enable** button to enable the use of Ingress.

<br />

#### Thanos SideCar

> [Thanos](https://prometheus-operator.dev/docs/operator/thanos/) is an open-source system for running large-scale, distributed, and highly available Prometheus setups. Thanos allows Prometheus to store data for extended periods by enabling data to be stored in object storage, such as Amazon S3 or Google Cloud Storage, instead of a local disk. This enables Prometheus to scale horizontally without needing to worry about running out of local storage space.

Reviewer: ```suggestion
[Thanos](https://prometheus-operator.dev/docs/operator/thanos/) is an open-source system for running large-scale, distributed, and highly available Prometheus setups. Thanos allows Prometheus to store data for extended periods in object storage, such as Amazon S3 or Google Cloud Storage, instead of a local disk. This enables Prometheus to scale horizontally without the risk of using up local storage space.
```

<br />

#### Email Alerts

You can configure the Prometheus server to send email alerts to a set of contacts. Toggle the **Email Alerts** button to enable email alerting. Update the `alertmanager.config.receivers` settings with all the required email setting values.

<br />

```yaml
charts:
  kube-prometheus-stack:
    alertmanager:
      config:
        receivers:
          - name: email-alert
            email_configs:
              - to: <reciever>@<domainname>.com
                send_resolved: true
                from: <sender>@<domainname>.com
                smarthost: smtp.<domainname>.com:587
                auth_username: <sender>@<domainname>.com
                auth_identity: <sender>@<domainname>.com
                auth_password: <sender_passwd>
```

Refer to the [Prometheus Alertmanager Configuration](https://prometheus.io/docs/alerting/latest/configuration/) documentation to learn more about Alertmanager.

<br />

#### Grafana Ingress

You can enable an ingress endpoint for Grafana that will deploy an NGINX ingress controller. This feature can be used to enable HTTPS and require authentication for all Prometheus API requests. 

> By default, a service with a load balancer, exposing port 80 is created unless the ingress option is enabled. 

Reviewer: ```suggestion
If you do not enable the ingress option, then by default a service with a load balancer will be created that exposes port 80.

```

Toggle the **Enable** button to enable the use of Ingress.

<br />

#### Thanos SideCar

> [Thanos](https://prometheus-operator.dev/docs/operator/thanos/) is an open-source system for running large-scale, distributed, and highly available Prometheus setups. Thanos allows Prometheus to store data for extended periods by enabling data to be stored in object storage, such as Amazon S3 or Google Cloud Storage, instead of a local disk. This enables Prometheus to scale horizontally without needing to worry about running out of local storage space.

Reviewer: ```suggestion
[Thanos](https://prometheus-operator.dev/docs/operator/thanos/) is an open-source system for running large-scale, distributed, and highly available Prometheus setups. Thanos allows Prometheus to store data for extended periods in object storage, such as Amazon S3 or Google Cloud Storage, instead of a local disk. This enables Prometheus to scale horizontally without the risk of using up local storage space.

```

* [Prometheus](https://prometheus.io/)
* [Prometheus Alertmanager](https://www.prometheus.io/docs/alerting/latest/alertmanager/).
* [node-exporter](https://github.com/helm/charts/tree/master/stable/prometheus-node-exporter)
* [kube-state-metrics](https://github.com/helm/charts/tree/master/stable/kube-state-metrics)
* [Grafana](https://github.com/helm/charts/tree/master/stable/grafana)
* and the service monitors to scrape internal Kubernetes components.

</Tabs.TabPane>

<Tabs.TabPane tab="35.5.x" key="35.5.x">

## Prerequisites

* Kubernetes v1.16 or greater.

## Parameters

The Prometheus operator supports all the parameters exposed by the kube-prometheus-stack Helm Chart. Refer to the [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stackn) documentation for details.

The Prometheus Operator pack has one parameter you must initialize `grafana.adminPassword`:

<br />

```yaml
charts:
  kube-prometheus-stack:
    grafana:
      adminPassword: ""
```

Use the `grafana.adminPassword` parameter to assign a password to the Grafana admin user `admin`. 

> Additional parameters you should be aware of can be found by expanding the **Presets** view of the pack. You can modify the preset settings during the profile creation process or the cluster deployment process when reviewing the cluster profile.

Reviewer: ```suggestion
Additional parameters you should be aware of can be found by expanding the **Presets** options. You can modify the preset settings when you create the profile or when you deploy the cluster and review the cluster profile.

```

<br />

#### Email Alerts

You can configure the Prometheus server to send email alerts to a set of contacts. Toggle the **Email Alerts** button to enable email alerting. Update the `alertmanager.config.receivers` settings with all the required email setting values.

<br />

```yaml
charts:
  kube-prometheus-stack:
    alertmanager:
      config:
        receivers:
          - name: email-alert
            email_configs:
              - to: <reciever>@<domainname>.com
                send_resolved: true
                from: <sender>@<domainname>.com
                smarthost: smtp.<domainname>.com:587
                auth_username: <sender>@<domainname>.com
                auth_identity: <sender>@<domainname>.com
                auth_password: <sender_passwd>
```

Refer to the [Prometheus Alertmanager Configuration](https://prometheus.io/docs/alerting/latest/configuration/) documentation to learn more about Alertmanager.

<br />

#### Grafana Ingress

You can enable an ingress endpoint for Grafana that will deploy an NGINX ingress controller. This feature can be used to enable HTTPS and require authentication for all Prometheus API requests. 

> By default, a service with a load balancer, exposing port 80 is created unless the ingress option is enabled. 

Reviewer: ```suggestion
If you do not enable the ingress option, then by default a service with a load balancer will be created that exposes port 80.

```

Toggle the **Enable** button to enable the use of Ingress.

<br />

#### Thanos SideCar

> [Thanos](https://prometheus-operator.dev/docs/operator/thanos/) is an open-source system for running large-scale, distributed, and highly available Prometheus setups. Thanos allows Prometheus to store data for extended periods by enabling data to be stored in object storage, such as Amazon S3 or Google Cloud Storage, instead of a local disk. This enables Prometheus to scale horizontally without needing to worry about running out of local storage space.

Reviewer: ```suggestion
[Thanos](https://prometheus-operator.dev/docs/operator/thanos/) is an open-source system for running large-scale, distributed, and highly available Prometheus setups. Thanos allows Prometheus to store data for extended periods in object storage, such as Amazon S3 or Google Cloud Storage, instead of a local disk. This enables Prometheus to scale horizontally without the risk of using up local storage space.

```

* [Prometheus](https://prometheus.io/)
* [Prometheus Alertmanager](https://www.prometheus.io/docs/alerting/latest/alertmanager/).
* [node-exporter](https://github.com/helm/charts/tree/master/stable/prometheus-node-exporter)
* [kube-state-metrics](https://github.com/helm/charts/tree/master/stable/kube-state-metrics)
* [Grafana](https://github.com/helm/charts/tree/master/stable/grafana)
* and the service monitors to scrape internal Kubernetes components.

</Tabs.TabPane>

<Tabs.TabPane tab="30.2.x" key="30.2.x">

## Prerequisites

* Kubernetes v1.16 or greater.

## Parameters

The Prometheus operator supports all the parameters exposed by the kube-prometheus-stack Helm Chart. Refer to the [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stackn) documentation for details.

The Prometheus Operator pack has one parameter you must initialize `grafana.adminPassword`:

<br />

```yaml
charts:
  kube-prometheus-stack:
    grafana:
      adminPassword: ""
```

Use the `grafana.adminPassword` parameter to assign a password to the Grafana admin user `admin`. 

> Additional parameters you should be aware can be found by expanding the **Presets** view of the pack. You can modify the preset settings during the profile creation process or the cluster deployment process when reviewing the cluster profile.

Reviewer: ```suggestion
Additional parameters you should be aware can be found by expanding the **Presets** options. You can modify the preset settings when you create the profile creation or when you deploy the cluster and review the cluster profile.

```

<br />

#### Email Alerts

You can configure the Prometheus server to send email alerts to a set of contacts. Toggle the **Email Alerts** button to enable email alerting. Update the `alertmanager.config.receivers` settings with all the required email setting values.

<br />

```yaml
charts:
  kube-prometheus-stack:
    alertmanager:
      config:
        receivers:
          - name: email-alert
            email_configs:
              - to: <reciever>@<domainname>.com
                send_resolved: true
                from: <sender>@<domainname>.com
                smarthost: smtp.<domainname>.com:587
                auth_username: <sender>@<domainname>.com
                auth_identity: <sender>@<domainname>.com
                auth_password: <sender_passwd>
```

Refer to the [Prometheus Alertmanager Configuration](https://prometheus.io/docs/alerting/latest/configuration/) documentation to learn more about Alertmanager.

<br />

#### Grafana Ingress

You can enable an ingress endpoint for Grafana that will deploy an NGINX ingress controller. This feature can be used to enable HTTPS and require authentication for all Prometheus API requests. 

> By default, a service with a load balancer, exposing port 80 is created unless the ingress option is enabled. 

Reviewer: ```suggestion
If you do not enable the ingress option, then by default a service with a load balancer will be created that exposes port 80.

```
